src/NUnitFramework/tests/Internal/Filters/AndFilterTests.cs

```diff
@@ -71,6 +71,74 @@ public void ExplicitMatchTest()
         }
 
         [Test]
+        public void ExplicitMatchWithNotTest()
+        {
+            var filter = new AndFilter(
+                new NotFilter(
+                    new CategoryFilter("Dummy")),
+                new NotFilter(
+                    new IdFilter(_dummyFixture.Id)));
+
+            Assert.False(filter.IsExplicitMatch(_topLevelSuite));
+            Assert.False(filter.IsExplicitMatch(_dummyFixture));
+            Assert.False(filter.IsExplicitMatch(_dummyFixture.Tests[0]));
+
+            Assert.That(filter.Match(_anotherFixture), "4");
+        }
+
+        [TestCase(new bool[] { false, false}, false, StrictIdFilterForTests.EqualValueFunction.IsExplicitMatch)]
+        [TestCase(new bool[] { true, false }, false, StrictIdFilterForTests.EqualValueFunction.IsExplicitMatch)]
+        [TestCase(new bool[] { false, true }, false, StrictIdFilterForTests.EqualValueFunction.IsExplicitMatch)]
+        [TestCase(new bool[] { true, true }, true, StrictIdFilterForTests.EqualValueFunction.IsExplicitMatch)]
+        [TestCase(new bool[] { false, false }, false, StrictIdFilterForTests.EqualValueFunction.Match)]
+        [TestCase(new bool[] { true, false }, false, StrictIdFilterForTests.EqualValueFunction.Match)]
+        [TestCase(new bool[] { false, true }, false, StrictIdFilterForTests.EqualValueFunction.Match)]
+        [TestCase(new bool[] { true, true }, true, StrictIdFilterForTests.EqualValueFunction.Match)]
+        [TestCase(new bool[] { false, false }, false, StrictIdFilterForTests.EqualValueFunction.Pass)]
+        [TestCase(new bool[] { true, false }, false, StrictIdFilterForTests.EqualValueFunction.Pass)]
+        [TestCase(new bool[] { false, true }, false, StrictIdFilterForTests.EqualValueFunction.Pass)]
+        [TestCase(new bool[] { true, true }, true, StrictIdFilterForTests.EqualValueFunction.Pass)]
+        public void CombineTest(IEnumerable<bool> inputBooleans, bool expectedResult,
+            StrictIdFilterForTests.EqualValueFunction expectedCall)
+        {
+            var filters = new List<StrictIdFilterForTests>();
+            foreach (var inputBool in inputBooleans)
+            {
+                string id = string.Empty;
+                if (inputBool)
+                {
+                    id = _dummyFixture.Id;
+                }
+
+                var strictFilter = new StrictIdFilterForTests(id, expectedCall);
+                Assert.AreEqual(inputBool, Match(strictFilter, expectedCall));
+
+                filters.Add(strictFilter);
+            }
+
+            var filter = new AndFilter(filters.ToArray());
+            bool calculatedResult = Match(filter, expectedCall);
+            Assert.AreEqual(expectedResult, calculatedResult);
+        }
+
+        private bool Match(
```

Comment:
Having a test method with the same name as one of the underlying TestFilter methods is a bit confusing.